src/core/MOM.F90

```diff
@@ -2339,8 +2341,13 @@ subroutine register_diags(Time, G, GV, CS, ADp)
 
   thickness_units = get_thickness_units(GV)
   flux_units      = get_flux_units(GV)
-  T_flux_units    = get_tr_flux_units(GV, "degC")
-  S_flux_units    = get_tr_flux_units(GV, "psu")
+  T_flux_units    = trim("Watts/m2")
+  S_flux_units    = get_tr_flux_units(GV, "PPT")
+  if (.not. GV%Boussinesq) then
+    call MOM_error(WARNING, "register_diags: Conversion from temperature flux to W m2 for advective" // &
```

Comment:
The conversion for non-Boussinesq is correct. I'll remove this warning in the merge too.

Reply:
Great. I thought the model was properly setup to do non-Bouss seamlessly.